fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java

```diff
@@ -507,6 +508,31 @@ private LogicalJoin<Plan, Plan> bindJoin(MatchingContext<LogicalJoin<Plan, Plan>
         LogicalJoin<Plan, Plan> join = ctx.root;
         CascadesContext cascadesContext = ctx.cascadesContext;
 
+        Set<String> tableNames = new HashSet<>();
+        List<Plan> children = join.children();
+        for (Plan child : children) {
+            // get alias name
+            if (child instanceof LogicalSubQueryAlias) {
+                LogicalSubQueryAlias subQueryAlias = (LogicalSubQueryAlias) child;
+                String alias = subQueryAlias.getAlias();
+
+                if (!tableNames.add(alias)) {
+                    throw new AnalysisException("Not unique table/alias: '" + alias + "'");
+                }
+                // get table name
+            } else if (child instanceof LogicalFilter) {
+                Plan grandChild = child.children().get(0);
+                if (grandChild instanceof LogicalOlapScan) {
```

Comment:
you should support other table type, for example, `LogicalFileScan` for hive table, so `LogicalRelation` is more suitable